nemo/collections/asr/parts/utils/nmse_clustering.py

```diff
@@ -388,21 +419,6 @@ def getPvalueList(self):
 
         return p_value_list
 
-    def getLamdaGaplist(self, lambdas):
-        lambdas = np.real(lambdas)
-        return list(lambdas[1:] - lambdas[:-1])
-
-    def estimateNumofSpeakers(self, affinity_mat):
-        """
-        Estimates the number of speakers using eigen decompose on Laplacian Matrix.
-        """
-        Laplacian = getLaplacian(affinity_mat)
-        lambdas, _ = eigDecompose(Laplacian, self.cuda)
-        lambdas = np.sort(lambdas)
-        lambda_gap_list = self.getLamdaGaplist(lambdas)
-        num_of_spk = np.argmax(lambda_gap_list[: min(self.max_num_speaker, len(lambda_gap_list))]) + 1
-        return num_of_spk, lambdas, lambda_gap_list
-
 
 def COSclustering(key, emb, oracle_num_speakers=None, max_num_speaker=8, min_samples=6, fixed_thres=None, cuda=False):
```

Comment:
min_samples: (int)
            Minimum number of samples required for NME clustering, this avoids
            zero p_neighbour_lists. Default of 6 is selected since  (1/rp_threshold) >= 4
            when max_rp_threshold = 0.25. Thus, NME analysis is skipped for matrices 
            smaller than (min_samples)x(min_samples).